test_suite/cloud/test_aws.py

```diff
@@ -558,12 +558,25 @@ def test_yum_group_install(self, host):
             pytest.skip('Not applicable to Atomic host AMIs')
 
         with host.sudo():
-            assert host.run_test('yum -y groupinstall "Development tools"'), \
-                'Error while installing Development tools group'
-
-            package_to_check = 'glibc-devel'
-            assert host.package(package_to_check).is_installed, \
-                f'{package_to_check} is not installed'
+            attempts = 2
+            for attempt in range(attempts):
+                try:
+                    assert host.run_test('yum -y groupinstall "Development tools"'), \
```

Comment:
If we need to check for the result first, we can avoid try/except to capture the `AssertionError` we produced by ourselves.

Try/except mechanism should not be used to control the flow of expected/known errors, but to handle unexpected situations instead.